e2e/nomostest/config_sync.go

```diff
@@ -245,6 +246,34 @@ func InstallConfigSync(nt *NT) error {
 	return nil
 }
 
+// InstallConfigSyncFromManifest installs ConfigSync on the test cluster by directly
+// applying the manifest file using kubectl client-side apply
+func InstallConfigSyncFromManifest(nt *NT) error {
```

Comment:
This will revert some of the modifications made by the usual [InstallConfigSync](https://github.com/GoogleContainerTools/config-sync/blob/main/e2e/nomostest/config_sync.go#L209) function used by the test scaffolding. Can this be consolidated to use the same function? I expect you could add an option to use `Update` instead of Patch/Apply

Reply:
I'm suggesting it would use the client-go `Update` (KubeClient.Update) method instead of the client-go `Apply` method (KubeClient.Apply)

Reply:
The go-client update has a few edge cases too.

When working on the configsync manifest, CRD like clusters.clusterregistry.k8s.io disallows update when there's no resourceversion attached. Something like 
```
    cli_test.go:1477: 2025-09-10 05:42:53.626931518 +0000 UTC ERROR: customresourcedefinitions.apiextensions.k8s.io "clusters.clusterregistry.k8s.io" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update
```

The operator also removes the admission webhook, which will need a special handle when restoring the shared test env, a create instead of update. We could also use preventDrift: true when configuring the configmanagement? 

Anyways not sure if the effort is worthy for a deprecated use case test.

Reply:
OK added the resourceVersion, hopefully it solves the issue